### ISSUE TITLE
My imput mask is 0

### ISSUE THREAD

**Issue by HaixiongLi** [milesial/Pytorch-UNet#228]
Hello, I use the kitti dataset, and mask is uint8 type, pixel value is 0 to 18, I set net.classes=20。
I change the code:
In train.py, I add this
`true_masks = true_masks.squeeze(1)`
before 
`loss = criterion(masks_pred, true_masks)`

In eval.py, I add this
`true_masks = true_masks.squeeze(1)`
before
`tot += F.cross_entropy(mask_pred, true_masks).item()`。
 
But I find when I use this code to train, the predict result is 0. 
So,I find in train.py, after this code, My true mask_type is all 0, before line 75, the value is pixel value/255.

`73                imgs = imgs.to(device=device, dtype=torch.float32)`
`74                mask_type = torch.float32 if net.n_classes == 1 else torch.long`
`75                true_masks = true_masks.to(device=device, dtype=mask_type)`

Excuse me, what is wrong with my code! 

Thanks!

**Comment by kirmans** [comment]
did you solve this problem ? @HaixiongLi

**Comment by stebbibg** [comment]
I had a similar problem. In dataset.py the images are normalized:

`        # HWC to CHW
        img_trans = img_nd.transpose((2, 0, 1))
        if img_trans.max() > 1:
            img_trans = img_trans / 255`

Since this code was made with binary masks and all mask values were between zero and 1, only the images were normalized from zero to one. If your masks include values over 1, which they do, the masks also get normalized and rounded to zero.

I had 5 classes and made a quick fix by setting the max value to 6 instead of one, since I could assume that I have no images with no pixel values over 6. Instead of checking the max value of the image, you can check if img_trans is a mask or an image before normalizing.

**Comment by stebbibg** [comment]
I had a similar problem. In dataset.py the images are normalized:
`# HWC to CHW`
`img_trans = img_nd.transpose((2, 0, 1)) `
`if img_trans.max() > 1:`
`    img_trans = img_trans / 255`

Since this code was made with binary masks and all mask values were between zero and 1, only the images were normalized from zero to one. If your masks include values over 1, which they do, the masks also get normalized and rounded to zero.

I had 5 classes and made a quick fix by setting the max value to 6 instead of one, since I could assume that I have no images with no pixel values over 6. Instead of checking the max value of the image, you can check if img_trans is a mask or an image before normalizing.
